src/http2_connection.erl

```diff
@@ -1458,8 +1457,87 @@ maybe_hpack(Continuation, Conn)
                continuation=undefined
               }}
     end;
+%% If not, we have to wait for all the CONTINUATIONS to roll in.
 maybe_hpack(Continuation, Conn) ->
     {next_state, continuation,
      Conn#connection{
        continuation = Continuation
       }}.
+
+%% Stream API: These will be moved
+-spec recv_h(
+        Stream :: h2_stream_set:stream(),
+        Conn :: connection(),
+        Headers :: hpack:headers()) ->
+                    ok.
+recv_h(Stream,
+       Conn,
+       Headers) ->
+    case h2_stream_set:type(Stream) of
```

Comment:
The compromise I had to make for it all to work is to leave anything that needs to understand #connection{} in here.